packages/web-components/src/switch/switch.styles.ts

```diff
@@ -0,0 +1,170 @@
+import { css } from '@microsoft/fast-element';
+import { display } from '@microsoft/fast-foundation';
+import {
+  colorCompoundBrandBackground,
+  colorCompoundBrandBackgroundHover,
+  colorCompoundBrandBackgroundPressed,
+  colorNeutralBackgroundDisabled,
+  colorNeutralForeground1,
+  colorNeutralForeground3,
+  colorNeutralForeground3Hover,
+  colorNeutralForeground3Pressed,
+  colorNeutralForegroundDisabled,
+  colorNeutralForegroundInverted,
+  colorNeutralForegroundInvertedHover,
+  colorNeutralForegroundInvertedPressed,
+  colorNeutralStrokeAccessible,
+  colorNeutralStrokeAccessibleHover,
+  colorNeutralStrokeAccessiblePressed,
+  colorNeutralStrokeDisabled,
+  colorPaletteRedForeground3,
+  colorTransparentBackground,
+  curveEasyEase,
+  durationNormal,
+  fontFamilyBase,
+  fontSizeBase300,
+  fontWeightRegular,
+  lineHeightBase300,
+  spacingHorizontalS,
+  spacingHorizontalXS,
+  spacingVerticalS,
+  spacingVerticalXS,
+} from '../theme/design-tokens.js';
+
+export const styles = css`
+  ${display('inline-flex')}
+
+  :host {
+    align-items: center;
+    flex-direction: row-reverse;
+    outline: none;
+    user-select: none;
+  }
+  :host([label-position='before']) {
+    flex-direction: row;
+  }
+  :host([label-position='above']) {
+    flex-direction: column;
+    align-items: flex-start;
+  }
+  :host([disabled]) .label,
+  :host([readonly]) .label,
+  :host([readonly]) .switch,
+  :host([disabled]) .switch {
+    cursor: not-allowed;
+  }
+  :host .label {
+    position: relative;
+    color: ${colorNeutralForeground1};
+    line-height: ${lineHeightBase300};
+    font-size: ${fontSizeBase300};
+    font-weight: ${fontWeightRegular};
+    font-family: ${fontFamilyBase};
+    padding: ${spacingVerticalXS} ${spacingHorizontalXS};
+    cursor: pointer;
+  }
+  :host([required]) .label:after {
+    content: '*';
```

Comment:
Do we want to use content to do this? How does this scale as it relates to "label" as a component?

Reply:
I'm not sure the best way to handle this. Do we expect that we are going to want to use the Label component in contexts such as this? If so then I would expect the Label to handle the `required` styling for the label specifically. Further, would we just want to set a `required` attribute on the Label itself? Or can we pass the required attribute from the Switch to the Label component. Just thinking out loud..

Reply:
Right... well I think we are probably expecting users to use the Label component in this instance? In which case I would expect the Label to handle the required styling.

Reply:
[Label spec](https://github.com/microsoft/fluentui/pull/26879)